WordPress/src/main/java/org/wordpress/android/ui/blaze/ui/blazewebview/BlazeWebViewFragment.kt

```diff
@@ -0,0 +1,204 @@
+import org.wordpress.android.ui.WPWebViewActivity.WPCOM_LOGIN_URL
+import org.wordpress.android.ui.blaze.BlazeActionEvent
+import org.wordpress.android.ui.blaze.BlazeWebViewHeaderUiState
+import org.wordpress.android.ui.blaze.BlazeWebViewClient
+import org.wordpress.android.ui.blaze.BlazeWebViewContentUiState
+import org.wordpress.android.ui.blaze.OnBlazeWebViewClientListener
+import org.wordpress.android.ui.blaze.ui.blazeoverlay.BlazeViewModel
+import org.wordpress.android.ui.compose.theme.AppTheme
+import org.wordpress.android.ui.compose.utils.withFullContentAlpha
+
+@AndroidEntryPoint
+class BlazeWebViewFragment: Fragment(), OnBlazeWebViewClientListener,
+    WPWebChromeClientWithFileChooser.OnShowFileChooserListener {
+    private var chromeClient: WPWebChromeClientWithFileChooser? = null
+    private val blazeWebViewViewModel: BlazeWebViewViewModel by viewModels()
+    private val blazeViewModel: BlazeViewModel by activityViewModels()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ) = ComposeView(requireContext()).apply {
+        setContent {
+            AppTheme {
+                BlazeWebViewScreen()
+            }
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initBackPressHandler()
+        initViewModel()
+        observeViewModel()
+    }
+
+    private fun initViewModel() {
+        blazeWebViewViewModel.start(blazeViewModel.promoteUiState.value, blazeViewModel.getSource())
+    }
+
+    private fun observeViewModel() {
+        blazeWebViewViewModel.actionEvents.onEach(this::handleActionEvents).launchIn(viewLifecycleOwner.lifecycleScope)
+    }
+
+    private fun handleActionEvents(actionEvent: BlazeActionEvent) {
+        when (actionEvent) {
+            is BlazeActionEvent.FinishActivity -> requireActivity().finish()
+            is BlazeActionEvent.LaunchExternalBrowser -> {
+                ActivityLauncher.openUrlExternal(
+                    requireContext(),
+                    actionEvent.url
+                )
+            }
+        }
+    }
+
+    // The next 2 Composable(s) live in the fragment because they need access to the chromeClient outside of the fun
+    // Also the clients needs access to activity and we are not holding on to that elsewhere
+    @Composable
+    private fun BlazeWebViewScreen(
+        viewModel: BlazeWebViewViewModel = androidx.lifecycle.viewmodel.compose.viewModel()
+    ) {
+        val data by viewModel.model.collectAsState()
+        val blazeHeaderState by viewModel.blazeHeaderState.collectAsState()
+        Scaffold(
+            topBar = { TopAppBar(blazeHeaderState) },
+            content = { BlazeWebViewContent(data) }
+        )
+    }
+
+    @SuppressLint("SetJavaScriptEnabled")
+    @Composable
+    private fun BlazeWebViewContent(model: BlazeWebViewContentUiState) {
```

Comment:
🔍 Nitpick: Can we move this under `TopAppBar`?. Just want to arrange the views according to the way they are rendered/called

Reply:
Absolutely. Good idea. In commit [9ea859b](https://github.com/wordpress-mobile/WordPress-Android/pull/18013/commits/9ea859b87670075c9f997cfd7a12b9420096a4cb)